### PR TITLE
8324750: C2: rename Matcher methods using "superword" -> "autovectorization"

### PR DESCRIPTION
**Pull request by eme64**
Subtask of https://github.com/openjdk/jdk/pull/16620

Matcher::superword_max_vector_size -> Matcher::max_vector_size_autovectorization

Matcher::match_rule_supported_superword -> match_rule_supported_autovectorization

This is to make the naming more general, since these methods can be used by any autovectorizer in the future.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8324750](https://bugs.openjdk.org/browse/JDK-8324750): C2: rename Matcher methods using "superword" -&gt; "autovectorization" (**Sub-task** - P4)

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)
 * [Christian Hagedorn](https://openjdk.org/census#chagedorn) (@chhagedorn - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17583/head:pull/17583` \
`$ git checkout pull/17583`

Update a local copy of the PR: \
`$ git checkout pull/17583` \
`$ git pull https://git.openjdk.org/jdk.git pull/17583/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17583`

View PR using the GUI difftool: \
`$ git pr show -t 17583`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17583.diff">https://git.openjdk.org/jdk/pull/17583.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17583#issuecomment-1911779790)

### CODE REVIEW

**Review by vnkozlov — APPROVED** [review]
Good and I think trivial.

File: src/hotspot/cpu/aarch64/aarch64.ad [hunk 1]
```diff
@@ -2377,7 +2377,7 @@ int Matcher::min_vector_size(const BasicType bt) {
   return MIN2(size, max_size);
 }
 
-int Matcher::superword_max_vector_size(const BasicType bt) {
+int Matcher::max_vector_size_autovectorization(const BasicType bt) {
```
Comment (chhagedorn) [inline src/hotspot/cpu/aarch64/aarch64.ad:2380]: Should we use two separate words like `auto_vectorization`?

Comment (eme64) [reply src/hotspot/cpu/aarch64/aarch64.ad]: Done :)

**Review by vnkozlov — APPROVED** [review]
Update is good.

**Review by chhagedorn — APPROVED** [review]
Thanks for the update, looks good!

**Comment by eme64** [comment]
Thanks for the reviews @chhagedorn @vnkozlov !
/integrate

**Pull request closed without merging**